### PR TITLE
Make edit a route

### PR DESCRIPTION
**Pull request by hidde**
* Editing now has its own route. It takes a parameter in which we specify which card is editable (it's always one at this point)
* New `EditMutationWrapper` component contains logic to do mutation, Save/Cancel buttons, the surrounding form, so that we don't need to repeat that code across cards
* New `EmptyCard` component is used for empty cards (for if user doesn't have content for that particular card)

Note that View Profile and Edit Profile are still in the same component, the Profile component. but depending on the route parameter one or more cards can be in edit mode.

### CODE REVIEW

File: src/components/profile/Profile.vue [hunk 1]
```diff
@@ -2,31 +2,14 @@
   <main class="profile container">
     <div
       :class="
-        'profile__section' + (this.editMode ? ' profile__section--editing' : '')
+        'profile__section' +
+          (this.editing && this.editing === 'personal-info'
```
Comment (Gregoor) [inline src/components/profile/Profile.vue]: could be simplified to `this.editing === 'personal-info'`

File: src/components/profile/Profile.vue [hunk 2]
```diff
@@ -2,31 +2,14 @@
...
-          funTitle,
-          picture,
-          location,
-          description,
-        }"
-        @toggle-edit-mode="toggleEditMode"
-      />
       <EditPersonalInfo
-        v-else
+        v-if="this.editing && this.editing === 'personal-info'"
```
Comment (Gregoor) [inline src/components/profile/Profile.vue]: same as above

File: src/components/profile/edit/EditMutationWrapper.vue [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+export default {
+  name: 'EditMutationWrapper',
+  props: {
+    initialValues: Object,
+    formName: String,
+    editVariables: Object,
+  },
+  methods: {
+    cancelEdit() {
```
Comment (Gregoor) [inline src/components/profile/edit/EditMutationWrapper.vue]: looks like this isn't called?

File: src/components/profile/Profile.vue (hunk 2 above)
Comment (hidde) [reply src/components/profile/Profile.vue]: Ah you are right, fixed!

File: src/components/profile/edit/EditMutationWrapper.vue (hunk 3 above)
Comment (hidde) [reply src/components/profile/edit/EditMutationWrapper.vue]: Good point, it's now a route, removed.

**Comment by hidde** [comment]
Thanks @Gregoor. I'll merge it in sometime today, we can (and probably should) iterate from here.

**Pull request merged**